### PR TITLE
PEP 3113 - Remove use of tuple parameter unpacking

### PR DESCRIPTION
**Pull request by rodrigc**
https://twistedmatrix.com/trac/ticket/8346

This command:

```
2to3 -f tuple_params
```

was useful for locating the places that need to be fixed.

### CODE REVIEW

File: twisted/words/service.py [hunk 1]
```diff
@@ -382,7 +382,8 @@ def logInAs(self, nickname, password):
          ":End of /MOTD command.")
         ]
 
-    def _cbLogin(self, (iface, avatar, logout)):
+    def _cbLogin(self, iface_avatar_logout):
```
Comment (adiroiban) [inline twisted/words/service.py]: maybe just call this `result` as this is the standard naming

File: docs/words/examples/oscardemo.py [hunk 2]
```diff
@@ -46,7 +46,8 @@ def gotBuddyList(self, l):
         self.setIdleTime(0)
         self.clientReady()
         self.createChat('%s Chat'%SN).addCallback(self.createdRoom)
-    def createdRoom(self, (exchange, fullName, instance)):
+    def createdRoom(self, exchange_fullname_instance):
```
Comment (adiroiban) [inline docs/words/examples/oscardemo.py]: this is a callback, so exchange_fullname_instance should be called just `result`

File: docs/words/examples/oscardemo.py [hunk 3]
```diff
@@ -66,7 +67,8 @@ def receiveMessage(self, user, multiparts, flags):
             self.lastUser = user.name
             self.sendMessage(user.name, multiparts, wantAck = 1, autoResponse = (self.awayMessage!=None)).addCallback( \
                 self.messageAck)
-    def messageAck(self, (username, message)):
+    def messageAck(self, username_message):
```
Comment (adiroiban) [inline docs/words/examples/oscardemo.py]: same... just a callback and we should call it 'result'

File: twisted/conch/insults/window.py [hunk 4]
```diff
@@ -386,10 +386,12 @@ def resize(self, width, height):
         if self.y >= height:
             self.y = height - 1
 
-    def __getitem__(self, (x, y)):
+    def __getitem__(self, x_y):
```
Comment (adiroiban) [inline twisted/conch/insults/window.py]: maybe, here we have a `key`

Comment (adiroiban) [inline twisted/conch/insults/window.py]: sorry ... `index`

File: twisted/conch/insults/window.py [hunk 5]
```diff
@@ -386,10 +386,12 @@ def resize(self, width, height):
...
+        (x, y) = x_y
         return self.contents[(self._width * y) + x]
 
-    def __setitem__(self, (x, y), value):
+    def __setitem__(self, x_y, value):
```
Comment (adiroiban) [inline twisted/conch/insults/window.py]: as above ... (self, index, value)

File: twisted/conch/insults/window.py [hunk 6]
```diff
@@ -420,7 +422,9 @@ def verticalLine(terminal, x, top, bottom):
     terminal.selectCharacterSet(insults.CS_US, insults.G0)
 
 
-def rectangle(terminal, (top, left), (width, height)):
+def rectangle(terminal, top_left, width_height):
```
Comment (adiroiban) [inline twisted/conch/insults/window.py]: top_left -> position
width_height -> size / dimension

File: twisted/conch/ssh/forwarding.py [hunk 7]
```diff
@@ -185,9 +185,11 @@ def connectionLost(self, reason):
             self.channel = None
 
 
-def packOpen_direct_tcpip((connHost, connPort), (origHost, origPort)):
+def packOpen_direct_tcpip(conn_host_port, orig_host_port):
```
Comment (adiroiban) [inline twisted/conch/ssh/forwarding.py]: conn_host_port -> destinationi
orig_host_port -> source

and we need a docstring for this, to declare the type and format of these arguments

File: twisted/conch/insults/window.py (hunk 6 above)
Comment (adiroiban) [inline twisted/conch/insults/window.py]: and we need docstring  to describe the arguments

File: twisted/conch/ssh/session.py [hunk 8]
```diff
@@ -323,11 +323,12 @@ def parseRequest_pty_req(data):
     modes = [(ord(modes[i]), struct.unpack('>L', modes[i+1: i+5])[0]) for i in range(0, len(modes)-1, 5)]
     return term, winSize, modes
 
-def packRequest_pty_req(term, (rows, cols, xpixel, ypixel), modes):
+def packRequest_pty_req(term, rows_cols_xpixel_ypixel, modes):
```
Comment (adiroiban) [inline twisted/conch/ssh/session.py]: rows_cols_xpixel_ypixel -> geometry  + docstring

File: twisted/conch/ssh/session.py [hunk 9]
```diff
@@ -341,9 +342,10 @@ def parseRequest_window_change(data):
     cols, rows, xpixel, ypixel = struct.unpack('>4L', data)
     return rows, cols, xpixel, ypixel
 
-def packRequest_window_change((rows, cols, xpixel, ypixel)):
+def packRequest_window_change(rows_cols_xpixel_ypixel):
```
Comment (adiroiban) [inline twisted/conch/ssh/session.py]: same as above we can call this argument geometry and document its format

File: twisted/conch/ssh/userauth.py [hunk 10]
```diff
@@ -187,12 +187,13 @@ def ssh_USERAUTH_REQUEST(self, packet):
         return d
 
 
-    def _cbFinishedAuth(self, (interface, avatar, logout)):
+    def _cbFinishedAuth(self, interface_avatar_logout):
```
Comment (adiroiban) [inline twisted/conch/ssh/userauth.py]: this is a callback... is we can go for `result`

File: twisted/mail/imap4.py [hunk 11]
```diff
@@ -1668,7 +1671,7 @@ def search_LARGER(self, query, id, msg):
     def search_NEW(self, query, id, msg):
         return '\\Recent' in msg.getFlags() and '\\Seen' not in msg.getFlags()
 
-    def search_NOT(self, query, id, msg, (lastSequenceId, lastMessageId)):
+    def search_NOT(self, query, id, msg, last_sequence_id_message_id):
```
Comment (adiroiban) [inline twisted/mail/imap4.py]: last_sequence_id_message_id -> `lastIDs` ... maybe

File: twisted/mail/pop3client.py [hunk 12]
```diff
@@ -109,17 +109,18 @@ def __init__(self, L):
         self.L = L
 
 
-    def setitem(self, (item, value)):
+    def setitem(self, item_value):
```
Comment (adiroiban) [inline twisted/mail/pop3client.py]: this is an ugly api... but maybe we can name this arg  `itemAndValue`

File: twisted/news/nntp.py [hunk 13]
```diff
@@ -398,7 +398,8 @@ def _passiveError(self, error):
         log.err('Passive Error: %s' % (error,))
 
 
-    def _headerInitial(self, (code, message)):
+    def _headerInitial(self, code_message):
```
Comment (adiroiban) [inline twisted/news/nntp.py]: maybe .... `response` or `line`

File: twisted/pair/test/test_rawudp.py [hunk 14]
```diff
@@ -11,7 +11,8 @@ class MyProtocol(protocol.DatagramProtocol):
     def __init__(self, expecting):
         self.expecting = list(expecting)
 
-    def datagramReceived(self, data, (host, port)):
+    def datagramReceived(self, data, host_port):
```
Comment (adiroiban) [inline twisted/pair/test/test_rawudp.py]: host_port -> peer

File: twisted/protocols/sip.py [hunk 15]
```diff
@@ -930,7 +931,7 @@ def handle_request(self, message, addr):
                     self.deliverResponse(self.responseFromRequest(e.code, message))
                 )
 
-    def handle_request_default(self, message, (srcHost, srcPort)):
+    def handle_request_default(self, message, src_host_port):
```
Comment (adiroiban) [inline twisted/protocols/sip.py]: maybe `sourcePeer` or just `peer`

File: twisted/words/service.py (hunk 1 above)
Comment (rodrigc) [inline twisted/words/service.py]: OK.

File: docs/words/examples/oscardemo.py (hunk 2 above)
Comment (rodrigc) [inline docs/words/examples/oscardemo.py]: OK.

File: docs/words/examples/oscardemo.py (hunk 3 above)
Comment (rodrigc) [inline docs/words/examples/oscardemo.py]: OK.

File: twisted/conch/insults/window.py (hunk 4 above)
Comment (rodrigc) [inline twisted/conch/insults/window.py]: OK.

File: twisted/conch/insults/window.py (hunk 5 above)
Comment (rodrigc) [inline twisted/conch/insults/window.py]: OK.

File: twisted/conch/insults/window.py (hunk 6 above)
Comment (rodrigc) [inline twisted/conch/insults/window.py]: OK.

File: twisted/conch/ssh/forwarding.py (hunk 7 above)
Comment (rodrigc) [inline twisted/conch/ssh/forwarding.py]: OK.

File: twisted/conch/ssh/session.py (hunk 9 above)
Comment (rodrigc) [inline twisted/conch/ssh/session.py]: OK.

File: twisted/conch/ssh/userauth.py (hunk 10 above)
Comment (rodrigc) [inline twisted/conch/ssh/userauth.py]: OK.

File: twisted/mail/imap4.py (hunk 11 above)
Comment (rodrigc) [inline twisted/mail/imap4.py]: OK.

File: twisted/mail/pop3client.py (hunk 12 above)
Comment (rodrigc) [inline twisted/mail/pop3client.py]: OK.

File: twisted/news/nntp.py (hunk 13 above)
Comment (rodrigc) [inline twisted/news/nntp.py]: OK.  I picked `response`

File: twisted/pair/test/test_rawudp.py (hunk 14 above)
Comment (rodrigc) [inline twisted/pair/test/test_rawudp.py]: OK.

File: twisted/protocols/sip.py (hunk 15 above)
Comment (rodrigc) [inline twisted/protocols/sip.py]: OK, I chose `sourcePeer`.

**Comment by rodrigc** [comment]
I incorporated your feedback and uploaded a new patch

**Comment by adiroiban** [comment]
Please don't rebase the branch associated with a PR as I would have to re-review the whole diff.

This time is fine as there are a lot of changes, but for the future please just continue to work on the initial branch submitted for review.

Thanks!

**Pull request merged**